engine/uState.mli

```diff
@@ -180,6 +180,7 @@ val merge_sort_context : ?loc:Loc.t -> sideff:bool -> rigid -> t -> UnivGen.sort
 val emit_side_effects : Safe_typing.private_constants -> t -> t
 
 val demote_global_univs : Environ.env -> t -> t
+val demote_global_univs_sound : Univ.ContextSet.t -> t -> t
```

Comment:
we should replace the old api not keep it

Reply:
sure, but we should also:
- have a look at inductives to see if the same API is ok
- change the API to declare constants to that i returns the uctx, right now I grab it via hook which is ugly

Reply:
About the second point: Dear @coq/vernac-maintainers I'd like to change the `declare.mli` API so that I can get my hands on the uctx below (tossed away by the `|> fst`):
```
let declare_definition_core ~info ~cinfo ~opaque ~obls ~body ?using sigma =
  let { CInfo.name; impargs; typ; _ } = cinfo in
  let entry, uctx = prepare_definition ~info ~opaque ?using ~name ~body ~typ sigma in
  let { Info.scope; clearbody; kind; hook; typing_flags; user_warns; ntns; _ } = info in
  let gref = declare_entry_core ~name ~scope ~clearbody ~kind ~impargs ~typing_flags ~user_warns ~obls ?hook ~uctx entry in
  List.iter (Metasyntax.add_notation_interpretation ~local:(info.scope=Locality.Discharge) (Global.env ())) ntns;
  gref, uctx
let declare_definition ~info ~cinfo ~opaque ~body ?using sigma =
  declare_definition_core ~obls:[] ~info ~cinfo ~opaque ~body ?using sigma |> fst
```
Options:
- export `declare_definition_core` (maybe `declare_definition_full` ?)
- export `prepare_definition` as other `prepare_*` available in the file and `declare_definition_entry` (the second part of the code above).

I'm not sure what would be best.

Reply:
>export declare_definition_core (maybe declare_definition_full ?)

I don't think we want to expose the `obls` argument, otherwise should be fine

Reply:
> we should replace the old api not keep it

I still use that function when I call 
```ocaml
  let gr, _ =
    if local then begin
      Dumpglob.dump_definition name true "var";
      comAssumption_declare_variable id Vernacexpr.NoCoercion ~kind (EConstr.to_constr sigma ty) ~univs ~impargs Glob_term.Explicit ~name
    end else begin
      Dumpglob.dump_definition name false "ax";
      comAssumption_declare_axiom Vernacexpr.NoCoercion ~local:Locality.ImportDefaultBehavior ~kind (EConstr.to_constr sigma ty)
        ~univs ~impargs ~inline:options.inline ~name ~id
    end
  in
```
The `_` is a `Univ.Instance.t`, so I don't think I can salvage that.

I tried to look at the code of `ComAssumption` but I have no idea which Ustate.t I can return

Reply:
you need the contextset which is sent to Global.push_context_set
it's in the `univs` argument (in the monomorphic case, when poly it's empty) which is untouched